tests/test_core/test_core.py

```diff
@@ -201,3 +208,39 @@ def test_opt_in_subcommand(
     # Check the contents of the config file match
     with open(config_file_path) as f:
         assert dict(yaml.safe_load(f)) == expected_configuration
+
+
+def test_unlock_state_lock_called(runner, mock_unlock):
```

Comment:
Do we need these tests in `test_core.py`? Feels more like a cli test.

Maybe mocking the result of `self.azure_storage.blob_exists` and test for the status message is more suitable in here? Open to any suggestions or ideas.

Reply:
You are right, these feel like cli tests. I have modified `test_core` to test `remove_state_lock` function as it is what is getting implemented in `core.py`. 

Good point about testing for  `self.azure_storage.blob_exists`, will add tests for this.